src/cosim/observer/file_observer.cpp

```diff
@@ -229,6 +233,59 @@ class file_observer::slave_value_writer
         }
     }
 
+    void create_metadata_file()
+    {
+        std::ofstream metadata_fw;
+        std::string filename;
+        std::stringstream ss;
+        constexpr int keyWidth = 14;
+
+        if (!timeStampedFileNames_) {
+            filename = observable_->name().append("_metadata.yaml");
+        } else {
+            auto time_str = format_time(boost::posix_time::microsec_clock::local_time());
+            filename = observable_->name().append("_").append(time_str).append("_metadata.yaml");
+        }
+
+        const auto filePath = logDir_ / filename;
+        metadata_fw.open(filePath, std::ios_base::out | std::ios_base::app);
+
+        if (fsw_.fail()) {
+            std::stringstream error;
+            error << "Failed to open log metadata file stream: " << filePath.c_str();
+            throw std::runtime_error(error.str());
+        }
+
+        auto md = observable_->model_description();
+
+        ss << std::left
+           << std::setw(keyWidth) << "name:" << md.name << std::endl
+           << std::setw(keyWidth) << "uuid:" << md.uuid << std::endl
+           << std::setw(keyWidth) << "description:" << md.description << std::endl
+           << std::setw(keyWidth) << "author:" << md.description << std::endl
+           << std::setw(keyWidth) << "version:" << md.version << std::endl;
+
+        ss << "variables:" << std::endl;
+
+        for (const auto& v : md.variables) {
```

Comment:
Instead of writing metadata for all variables, shouldn't this file reflect the content in the log file? In other words, use the logConfig information to only write data for logged variables?

Reply:
Absolutely, good catch. Updated accordingly now. A couple runs of `file_observer_logging_from_config_test` confirms that it works and metadata is only generated for configured variables.

Reply:
I also realized that in order for the CSV and YAML files to be "linkable", we have to do something, otherwise there would be no way to tell which metadata file corresponded to which CSV file when e.g. running multiple logging sessions through the same file observer. The simplest solution seems to be to just return the timestamp from create_log_file() and reuse it in create_metadata_file(), so the filenames output will look like this:

> slave uno_20221201_140801_676545.csv
> slave uno_20221201_140801_676545_metadata.yaml 

Alternatives would be creating a new directory for each logging, or linking them by specifying the CSV filename itself as metadata, but I think this solution should suffice.